File: haskell/IOHelpers.hs
module IOHelpers (clearScreen, onEnterContinue, printTextScreen, printBattlefield, delay) where

import Control.Concurrent (threadDelay)
import Drawer
import System.Process (callCommand)

clearScreen :: IO ()
clearScreen = callCommand "clear"

printTextScreen :: [String] -> IO ()
printTextScreen texts = do
  clearScreen
  mapM_ putStr (drawTextScreen texts)

printBattlefield ::
  [String] ->
  (Int, Int) ->
  (Int, Int) ->
  String ->
  [Int] ->
  [Int] ->
  IO ()
printBattlefield playerColors (playerLife, playerEnergy) (bossLife, bossEnergy) boss playerCards currentCards = do
  let battlefield = drawBattlefield playerColors (playerLife, playerEnergy) (bossLife, bossEnergy) boss playerCards currentCards
  clearScreen
  mapM_ putStr battlefield

onEnterContinue :: IO ()
onEnterContinue = do
  putStrLn " \n=== Pressione enter para continuar. ==="
  line <- getLine :: IO String
  putStrLn "\n\n"

delay :: IO ()
delay = threadDelay (1 * 1000000) -- 1.0 segundo

File: prolog/IOHelpers.pl
:- module(IOHelpers, [printStringList/1, printTextScreen/1, clearScreen/0, onEnterContinue/0, delay/0]).
:- use_module('./Helpers.pl').
:- use_module('./Drawer.pl').

clearScreen :-
  tty_clear.

onEnterContinue :-
  writeln(" \n=== Pressione enter para continuar. ==="),
  get_single_char(_),
  writeln("\n\n").

printStringList([]).
printStringList([Head|Tail]) :-
  write(Head),
  printStringList(Tail).

printTextScreen(Texts) :-
  clearScreen,
  drawTextScreen(Texts, TextScreen),
  maplist(write, TextScreen).

delay :-
    sleep(1).